pcs/src/univariate_kzg/succinct.rs

```diff
@@ -0,0 +1,67 @@
+use ark_ec::pairing::Pairing;
+use ark_ff::{BigInt, BigInteger, PrimeField, Zero};
+
+#[derive(Clone, Copy, Debug)]
+pub struct BnAffinePoint(pub BigInt<4>, pub BigInt<4>);
+
+pub fn wrap_g1affine(p: &G1Affine) -> BnAffinePoint {
+    BnAffinePoint(p.x.into_bigint(), p.y.into_bigint())
+}
+
+pub fn unwrap_g1affine(p: &BnAffinePoint) -> G1Affine {
+    G1Affine {
+        x: p.0.into(),
+        y: p.1.into(),
+        infinity: false,
+    }
+}
+
+pub fn msm(p: &[G1Affine], s: &[ScalarField]) -> G1Affine {
+    let mut iter = p.iter().zip(s).filter(|(_, s)| !s.is_zero());
+    let mut result = {
+        let (p, s) = iter.next().unwrap();
+        let mut p = wrap_g1affine(p);
+        bn254_double_and_add(&mut p, s);
```

Comment:
may i ask why we didn't use `AffinePoint` struct and the `::msm()` directly from `sp1-lib` as suggested in https://github.com/EspressoSystems/zkrollup-integration/issues/21#issuecomment-2292645127?

Reply:
I saw this after I had these working implementations. I don't think it's necessary to follow their APIs.